pip/src/interpreter.rs

```diff
@@ -174,6 +174,40 @@ impl Interpreter {
         ))
     }
 
+    /// Dumps the quantum state corresponding to the application of a given operation.
```

Comment:
I don't think this function belongs here. This file is supposed to be just interop "glue" and the code gets confusing when we start putting actual implementations in these sorts of files. Instead it should either go up or down a layer. Can we implement this in Python (in `_qsharp.py`) or in the native `Interpreter` and plumb it up?

Reply:
Fair point, I can move this down. That is what we would need to do anyway if we want to make this functionality available somewhere other than Python.

Reply:
I moved this down as you pointed out, and added unit tests at that layer.